libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -449,6 +454,82 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CWriteThenReadAsyncSuccess )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Test Function for I2C async read success
+ */
+TEST( TEST_IOT_I2C, AFQP_IotI2CReadAsyncAssisted )
+{
+    IotI2CHandle_t xI2CHandle;
+    int32_t lRetVal;
+    uint8_t ucReadBuf[16];
+    char cMsg[50]={0};
+
+    IotI2CConfig_t xI2CConfig = {
+        .ulBusFreq = testIotI2C_BAUDRATE,
+        .ulMasterTimeout =  testIotI2C_DEFAULT_TIMEOUT
+    };
+
+    if(TEST_PROTECT())
+    {
+        /* Open i2c to initialize hardware */
+        if (gIotI2cHandle[testIotI2C_INSTANCE] == NULL)
```

Comment:
gIotI2cHandle[testIotI2C_INSTANCE] will be always NULL in TEST_SETUP. So just open it directly.

Reply:
Why do you have this global variable if it does not keep information between tests?

Reply:
I remember gIotI2cHandle was added by your team in case the peripheral is already open by some code outside common i/o test. 

I handled that by explicitly close the peripheral first in test setup.

Reply:
done